### PR TITLE
Fixes in fs.watch and in getPrivateEnvVariables to allow browserAuth run in env with process defined

### PR DESCRIPTION
**Pull request by martin-lysk**
fixes watching on root folder, fixes watcher when multiple listener are attached, fixes private env in env where process is defined but

### CODE REVIEW

File: lix/source-code/fs/src/implementations/memoryFs.ts [hunk 1]
```diff
@@ -183,24 +183,30 @@ export function createNodeishMemoryFs(): NodeishFilesystem {
 			})
 
 			const listener = (event: FileChangeInfo) => {
-				if (event.filename === null) {
+				const eventToForward = {
+					eventType: event.eventType,
+					filename: event.filename,
+				}
+
+				if (eventToForward.filename === null) {
```
Comment (janfjohannes) [inline lix/source-code/fs/src/implementations/memoryFs.ts:191]: @martin-lysk  could you rename eventToForward to event and  just destructure the incoming event directly? into eventType, filename ertc?

**Review by janfjohannes — CHANGES_REQUESTED** [review]

**Comment by janfjohannes** [comment]
@martin-lysk  also needs manual rebase due to conflicts

File: lix/source-code/client/src/helpers.ts [hunk 2]
```diff
@@ -18,7 +18,8 @@ export const withLazyFetching = ({
 }: Args): NodeishFilesystem => {
 	return new Proxy(nodeishFs, {
 		get(getTarget: typeof nodeishFs, prop, receiver) {
-			if (getTarget[prop as keyof typeof nodeishFs]) {
+			// eslint-disable-next-line no-prototype-builtins
+			if (((getTarget as any).__proto__ as any).hasOwnProperty(prop)) {
```
Comment (janfjohannes) [inline lix/source-code/client/src/helpers.ts:22]: not sure i want this in general, why is this needed except for your testing?

**Review by janfjohannes — CHANGES_REQUESTED** [review]

**Review by janfjohannes — CHANGES_REQUESTED** [review]
sorry i forgot to submit this time. could you have a look at the protoype handling? i would not want to have to think if something is part of a prototype chain or a high level feature . i would expect i can replace any function call with the proxy.

File: lix/source-code/client/src/helpers.ts (hunk 2 above)
Comment (martin-lysk) [reply lix/source-code/client/src/helpers.ts:22]: @janfjohannes This problem occured when you subclass a nodish fs and mark a method as private - instead of the functional approach since private functions become visible to the target. The Proxy should only proxy methods that are defined in the NodishFs type but this is not the case with the current implementation.

Comment (janfjohannes) [reply lix/source-code/client/src/helpers.ts:22]: @martin-lysk  not sure what you mean, lets discuss in a call.

**Comment by samuelstroschein** [comment]
What is the status?

**Comment by janfjohannes** [comment]
@samuelstroschein  martin tries to have a look at my comments today.

**Review by janfjohannes — APPROVED** [review]
